apisix/core/dns/client.lua

```diff
@@ -130,7 +132,22 @@ end
 
 
 function _M.new(opts)
-    opts.ipv6 = true
+    local local_conf = config_local.local_conf()
+
+    if opts.ipv6 == nil then
```

Comment:
I spend some time digging into it. Look like it should be `enable_ipv6`, and it only works with the nameserver. See http://kong.github.io/lua-resty-dns-client/modules/resty.dns.client.html#init

This field is incorrect from the beginning...

Reply:
🤣 done